sample/legacy_with_buckets.go

```diff
@@ -0,0 +1,35 @@
+package sample
+
+import (
+	metrics "github.com/rcrowley/go-metrics"
+)
+
+type LegacyWithBuckets interface {
+	metrics.Sample
+	BucketsAndValues() (buckets []float64, values []int64)
+}
+
+type legacyWithBuckets struct {
+	metrics.Sample
+	bucketPartialSample
+}
+
+func NewLegacyWithBuckets(reservoirSize int, buckets ...float64) metrics.Sample {
```

Comment:
though for now, default pre-defined buckets are set as ints, I decided to keep floats here to to avoid unnecessary conversions.